E2E-Demos/Question-Answering/02.serve-vectorstore.ipynb

```diff
@@ -0,0 +1,293 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e22867e3-a69c-488a-819e-cced462be9e3",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import base64\n",
+    "import logging\n",
+    "import warnings\n",
+    "import subprocess\n",
+    "\n",
+    "import mlflow\n",
+    "import requests\n",
+    "import boto3\n",
+    "import s3transfer\n",
+    "import ipywidgets as widgets\n",
+    "\n",
+    "\n",
+    "warnings.filterwarnings('ignore')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b2c72b99-3585-49e9-993a-81764e7773a8",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Get all loggers\n",
+    "loggers = logging.Logger.manager.loggerDict.values()\n",
+    "\n",
+    "# Iterate over all loggers and set their level to ERROR\n",
+    "# as we don't want to polute the output of the code cells\n",
+    "# with debugging messages.\n",
+    "for logger in loggers:\n",
+    "    if isinstance(logger, logging.Logger):\n",
+    "        logger.setLevel(logging.ERROR)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7656fd38-0660-402b-bd83-72e566cd4e0f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import base64\n",
+    "\n",
+    "def encode_base64(message: str):\n",
+    "    encoded_bytes = base64.b64encode(message.encode('ASCII'))\n",
+    "    return encoded_bytes.decode('ASCII')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "36a8b07a-15d8-44a2-89c4-266ee61d1a1e",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# Logging the Vector Store as an Artifact\n",
+    "\n",
+    "To kick-off this process, you need to create a new experiment (or re-use and existing one) and log the Chroma DB files as an artifact of this experiment. In the end, you'll need to retrieve the URI pointing to the location of this artifact and pass it to the custom predictor component. This way, the custom predictor component will know hot to retrieve the artifact and serve it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5734b081-b53a-4c6c-a123-732229d25a54",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "EZAF_ENV = input(\"Enter your EzAF domain (e.g., hpe-ezaf): \")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4e23d780-57e0-4d5c-a699-aa3046a5b586",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_or_create_experiment(exp_name):\n",
+    "    \"\"\"Register an experiment in MLFlow.\n",
+    "    \n",
+    "    args:\n",
+    "      exp_name (str): The name of the experiment.\n",
+    "    \"\"\"\n",
+    "    try:\n",
+    "        mlflow.set_tracking_uri(f\"https://mlflow.{EZAF_ENV}.com\")\n",
```

Comment:
this is not needed, and EZAF_ENV is not needed as well, the tracking uri is set automatically in the backend